### ISSUE TITLE
Support .Args.XYZ instead of stencil.Arg "XYZ"

### ISSUE THREAD

**Issue by deregtd** [rgst-io/stencil#48]
Args are so common, for quality-of-life, just fill them into the base go-template vars under Args or Arg

**Comment by jaredallard** [comment]
Yeah, this makes sense to do. I think we probably need to move the JSON schema validation to before render time as well.

**Comment by jaredallard** [comment]
Hmmmm, I'm not sure this is a good idea. That would break the default logic, which would make dealing with arguments significantly more annoying. For example, `stencil.Arg "a.b.c"` would parse the JSON schema and set the default value for that type if it's not set at accessing time. We could retain that behaviour by building that entire map based on the JSON schema, but that feels much less clean for little benefit.

I'll still consider this, but I'm not sure if it's worth it.

**Comment by deregtd** [comment]
Oh are there defaults supported by the schema?  I'm not sure I've ever used those...  I would expect that defaults are made through usage in templates, not the schema.  (i.e. `if a.b.c != nil`)

**Comment by jaredallard** [comment]
> Oh are there defaults supported by the schema? I'm not sure I've ever used those... I would expect that defaults are made through usage in templates, not the schema. (i.e. `if a.b.c != nil`)

Hmmm, that's fair. I guess most people would be more familiar with [`default` from sprig](https://masterminds.github.io/sprig/defaults.html#:~:text=values%20for%20templates.-,default,-To%20set%20a), also.

**Comment by deregtd** [comment]
Yeah that's the system I used several places.  Basically a COALESCE() operator.

**Comment by jaredallard** [comment]
Migrated: https://git.rgst.io/rgst-io/stencil/issues/53
